### PR TITLE
Increase UT test coverage for api server/etcd data partition - part 1

### PR DESCRIPTION
**Pull request by Sindica**
Issue https://github.com/futurewei-cloud/arktos/issues/568

### CODE REVIEW

File: pkg/registry/core/controllerinstance/storage/storage_test.go [hunk 1]
```diff
@@ -118,3 +119,40 @@ func TestList(t *testing.T) {
...
+	defer server.Terminate(t)
+	defer storage.DestroyFunc()
+	controllerInstance := validNewControllerInstance()
+	ctx := genericapirequest.NewContext()
+	_, err := storage.Store.Create(ctx, controllerInstance, rest.ValidateAllObjectFunc, &metav1.CreateOptions{})
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+
+	object, err := storage.Get(ctx, "foo", &metav1.GetOptions{})
```
Comment (chenqianfzh) [inline pkg/registry/core/controllerinstance/storage/storage_test.go:134]: Let's define "foo" as a const var outside any funcs.

File: pkg/registry/core/controllerinstance/storage/storage_test.go [hunk 2]
```diff
@@ -118,3 +119,40 @@ func TestList(t *testing.T) {
...
+	}
+
+	// read back and compare
+	newObj, err := storage.Get(ctx, "foo", &metav1.GetOptions{})
+	if err != nil {
+		t.Errorf("unexpected error: %v", err)
+	}
+	newInstance := newObj.(*api.ControllerInstance)
+	if newInstance.ControllerKey != math.MaxInt64 {
+		t.Fatalf("controller key update failed")
```
Comment (chenqianfzh) [inline pkg/registry/core/controllerinstance/storage/storage_test.go:155]: Log the real newInstance.ControllerKey in the error message.

File: pkg/master/controller_test.go [hunk 3]
```diff
@@ -1290,3 +1291,112 @@ func TestCreateOrUpdateMasterService(t *testing.T) {
...
+				}
+			}
+
+			r := reconcilers.NewMasterCountEndpointReconciler(3, clientset.CoreV1())
+			err := r.RemoveEndpoints(test.serviceName, "", net.ParseIP(test.ip), test.endpointPorts)
+			if err != nil {
+				t.Errorf("case %q: unexpected error: %v", test.testName, err)
+			}
+			actualEndpoints, err := clientset.CoreV1().Endpoints(corev1.NamespaceDefault).Get(test.serviceName, metav1.GetOptions{})
+			if test.expectedErr != nil {
```
Comment (chenqianfzh) [inline pkg/master/controller_test.go:1390]: Need to add a test for case where test.expected == nil but err != nil

File: pkg/registry/core/controllerinstance/storage/storage_test.go (hunk 2 above)
Comment (Sindica) [reply pkg/registry/core/controllerinstance/storage/storage_test.go:155]: Should the previous value which is 1. But I updated it anyway.

**Pull request merged**